ta4j-core/src/main/java/org/ta4j/core/indicators/RWIHighIndicator.java

```diff
@@ -57,25 +57,28 @@ public RandomWalkIndexHighIndicator(TimeSeries series, int timeFrame) {
 
     @Override
     protected Decimal calculate(int index) {
-        int lastIndex = Math.max(0, index - timeFrame+1);
-        int n = 2;
-        AverageTrueRangeIndicator averageTrueRange = new AverageTrueRangeIndicator(getTimeSeries(), n);
-        Decimal highestRWI = maxPrice.getValue(index).minus(minPrice.getValue(Math.max(0, index-1)))
-                .dividedBy(averageTrueRange.getValue(index).multipliedBy(Decimal.valueOf(Math.sqrt(n))));
-
-        for(int i = index-2; i >= lastIndex; i--) {
-            n++;
-            averageTrueRange = new AverageTrueRangeIndicator(getTimeSeries(), n);
-            Decimal currentRWI = maxPrice.getValue(index).minus(minPrice.getValue(i))
-                    .dividedBy(averageTrueRange.getValue(index).multipliedBy(Decimal.valueOf(Math.sqrt(n))));
-            if(currentRWI.isGreaterThan(highestRWI)){
-                highestRWI = currentRWI;
-            }
-
+        Decimal highestRWI = Decimal.NaN;
+        
+        for(int n = 2; n <= this.timeFrame; n++) {
+           Decimal currentRWI = calcRWIHighValue(index, index-n, n);
+           if(currentRWI.isGreaterThan(highestRWI)){
```

Comment:
Will be always "false" because of "highestRwi" is Decimal.NaN.
Hence, you need: 

```
if (currentRWI.isGreaterThan(highestRWI) || highestRWI.isNaN()) {
	highestRWI = currentRWI;
}
```

The same for RwiLowIndicator.